packages/kit/src/runtime/client/utils.js

```diff
@@ -26,34 +26,62 @@ export function scroll_state() {
 
 const warned = new WeakSet();
 
+/** @typedef { 'preload-code' | 'preload-data' | 'noscroll' | 'reload' } AnchorShortAttribute */
+/** @typedef { `data-sveltekit-${AnchorShortAttribute}`} AnchorAttribute */
+
+const anchor_attribute_valid_values = /** @type {const} */ ({
+	'data-sveltekit-preload-code': ['', 'off', 'tap', 'hover', 'viewport', 'page'],
+	'data-sveltekit-preload-data': ['', 'off', 'tap', 'hover'],
+	'data-sveltekit-noscroll': ['', 'off'],
+	'data-sveltekit-reload': ['', 'off']
+});
+
+/** @typedef {typeof anchor_attribute_valid_values['data-sveltekit-preload-data'][number]} PreloadDataValidValues */
+/** @typedef {typeof anchor_attribute_valid_values['data-sveltekit-preload-code'][number]} PreloadCodeValidValues */
+/** @typedef {typeof anchor_attribute_valid_values['data-sveltekit-noscroll'][number]} NoscrollValidValues */
+/** @typedef {typeof anchor_attribute_valid_values['data-sveltekit-reload'][number]} ReloadValidValues */
+
 /**
  * @param {Element} element
- * @param {string} name
+ * @param {T} attributeName
+ * @returns {typeof anchor_attribute_valid_values[T][number] | null}
+ * @template {AnchorAttribute} T
+ */
+function getValidatedAttribute(element, attributeName) {
+	const value = element.getAttribute(attributeName);
+	return validateAttributeValue(element, attributeName, value);
```

Comment:
Does this file be bundled by esbuild or rollup at build time? The latter [doesn't treeshake](https://rollupjs.org/repl/?version=3.5.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtb2R1bGVfMS5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJmdW5jdGlvbiUyMGZvbyh4KSUyMCU3QiU1Q24lMjAlMjAlMjBpZiUyMChmYWxzZSklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIweCUyMCUyQiUzRCUyMDElM0IlNUNuJTIwJTIwJTIwJTdEJTVDbiUyMCUyMHJldHVybiUyMHglM0IlNUNuJTdEJTVDbiU1Q25leHBvcnQlMjBjb25zdCUyMHklMjAlM0QlMjBmb28oMiklM0IlMjIlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIybW9kdWxlXzIuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=) identity function calls so doing it this way might save some bytes.

```suggestion
	return __SVELTEKIT_DEV__ ? validateAttributeValue(element, attributeName, value) : value;
```

and the function below can skip the check.

Reply:
Good call -- I'm not sure it'd be treeshaken either way, but this makes more sense regardless.